src/Symfony/Component/Security/Http/Authenticator/Passport/SelfValidatingPassport.php

```diff
@@ -25,11 +26,20 @@
 class SelfValidatingPassport extends Passport
 {
     /**
+     * @param UserBadge        $userBadge
      * @param BadgeInterface[] $badges
      */
-    public function __construct(UserInterface $user, array $badges = [])
+    public function __construct($userBadge, array $badges = [])
     {
-        $this->user = $user;
+        if ($userBadge instanceof UserInterface) {
+            trigger_deprecation('symfony/security-http', '5.2', 'The user argument of "%s" must be a "%s" or a string containing the username.', __CLASS__, UserBadge::class);
```

Comment:
Given it was experimental in 5.1, is it really worth adding a BC Layer?

Reply:
I have a mixed feeling about this one. I think one should accept/expect BC breaks for experimental features. If it's well documented in the CHANGELOG, I think we should remove the deprecation notice and fail hard.

Reply:
Hmm, as this class is used in every custom authenticator made by early-adapters and the BC layer was very small, I decided to avoid the hard break. Would it help any if we remove the BC break in 5.3 already? (and adding that as part of the error message)

Reply:
@wouterj your plan sounds good to me.